fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -364,19 +377,33 @@ public class LuceneIndexTest extends FDBRecordStoreTestBase {
             LuceneIndexTypes.LUCENE,
             ImmutableMap.of());
 
-    private static final Index JOINED_INDEX = new Index("joinNestedConcat", concat(
-            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
-            field("simple").nest(function(LuceneFunctionNames.LUCENE_TEXT, field("text"))
-            )).groupBy(field("complex").nest("group")), LuceneIndexTypes.LUCENE,
-            ImmutableMap.of(INDEX_PARTITION_BY_TIMESTAMP, "complex.timestamp",
-                            INDEX_PARTITION_HIGH_WATERMARK, "10"));
+    private static final Index JOINED_INDEX = getJoinedIndex("10");
 
-    private static final Index JOINED_INDEX_NOGROUP = new Index("joinNestedConcat", concat(
-            field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
-            field("simple").nest(function(LuceneFunctionNames.LUCENE_TEXT, field("text"))
-            )), LuceneIndexTypes.LUCENE, ImmutableMap.of(
-                    INDEX_PARTITION_BY_TIMESTAMP, "complex.timestamp",
-                    INDEX_PARTITION_HIGH_WATERMARK, "10"));
+    @Nonnull
+    private static Index getJoinedIndex(final String partitionHighWatermark) {
+        return new Index("joinNestedConcat",
+                concat(
+                        field("complex").nest(function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
+                        field("simple").nest(function(LuceneFunctionNames.LUCENE_TEXT, field("text"))),
+                        field("complex").nest(function(LuceneFunctionNames.LUCENE_SORTED, field("timestamp")))
```

Comment:
The `timestamp` field was left out in both of these. I think this is necessary for partitioning to actually work, although I added the field before fixing the `getOldestNDocument` search.